src/compiler/python_generator.h

```diff
@@ -48,22 +58,72 @@ struct GeneratorConfiguration {
   grpc::string beta_package_root;
 };
 
-class PythonGrpcGenerator : public grpc::protobuf::compiler::CodeGenerator {
- public:
-  PythonGrpcGenerator(const GeneratorConfiguration& config);
-  ~PythonGrpcGenerator();
+// An abstract interface representing a method.
+struct Method {
+  virtual ~Method() {}
+
+  virtual grpc::string name() const = 0;
+
+  virtual grpc::string input_type_name() const = 0;
+  virtual grpc::string output_type_name() const = 0;
+  virtual bool get_module_message_path_input(grpc::string* out) const = 0;
+  virtual bool get_module_message_path_output(grpc::string* out) const = 0;
+
+  virtual bool ClientStreaming() const = 0;
+  virtual bool ServerStreaming() const = 0; 
+};
+
+// An abstract interface representing a service
+struct Service {
+  virtual ~Service() {}
+
+  virtual grpc::string name() const = 0;
+
+  virtual int method_count() const = 0;
+  virtual std::unique_ptr<const Method> method(int i) const = 0;
```

Comment:
FYI we're trying to move away from C++11 features in the generator code to get to building on Mac OS X 10.6 for Python.